### PR TITLE
feat: added autoscaling and ml node

### PR DESCRIPTION
**Pull request by tomhobson**

### CODE REVIEW

File: terraform-jx-cluster-aks/cluster/variables.tf [hunk 1]
```diff
@@ -8,12 +45,6 @@ variable "cluster_version" {
 variable "location" {
   type = string
 }
-variable "node_size" {
-  type = string
-}
-variable "node_count" {
```
Comment (hervelemeur) [inline terraform-jx-cluster-aks/cluster/variables.tf:14]: No `node_count` anymore?

**Review by hervelemeur — CHANGES_REQUESTED** [review]
Not sure the `node_count` should be removed.

File: terraform-jx-cluster-aks/cluster/variables.tf (hunk 1 above)
Comment (tomhobson) [reply terraform-jx-cluster-aks/cluster/variables.tf:14]: I just thought the less variables the better to be honest.

Which value do you want to be more important than the other?


If node_count is null then autoscale?
or if max_node_count and min_node_count are null then don't autoscale?

Comment (hervelemeur) [reply terraform-jx-cluster-aks/cluster/variables.tf:14]: For people not wanting autoscale/wanting to keep it simple, `node_count` is all they'll need.

> If node_count is null then autoscale?
or if max_node_count and min_node_count are null then don't autoscale?

Yeah I was thinking something like that :)

Comment (tomhobson) [reply terraform-jx-cluster-aks/cluster/variables.tf:14]: Done :+1:

**Comment by hervelemeur** [comment]
@tomhobson is it normal that the min and max (ml) nodes counts don't have the same default value in terraform-jx-cluster-aks/variables.tf and in variables.tf?

**Comment by tomhobson** [comment]
@hervelemeur fixed!

**Pull request merged**